### ISSUE TITLE
profile.dat parser crash

### ISSUE THREAD

**Issue by neuroscr** [loki-project/loki-network#459]
So looks like the data in my profiles.dat is crashing lokinet
`-rw-r--r-- 1 root root 337 Jan  9 00:22 /root/.lokinet/profiles.dat`

got an strace:
```
mprotect(0x7fa5f7fde000, 8388608, PROT_READ|PROT_WRITE) = 0
clone(child_stack=0x7fa5f87dcfb0, flags=CLONE_VM|CLONE_FS|CLONE_FILES|CLONE_SIGHAND|CLONE_THREAD|CLONE_SYSVSEM|CLONE_SETTLS|CLONE_PARENT_SETTID|CLONE_CHILD_CLEARTID, parent_tidptr=0x7fa5f87dd9d0, tls=0x7fa5f87dd700, child_tidptr=0x7fa5f87dd9d0) = 60916
futex(0x7fa5fa957150, FUTEX_WAIT_PRIVATE, 0, NULL) = 0
futex(0x7fa5fa957d50, FUTEX_WAKE_PRIVATE, 1) = 1
mmap(NULL, 8392704, PROT_NONE, MAP_PRIVATE|MAP_ANONYMOUS|MAP_STACK, -1, 0) = 0x7fa5f77dc000
mprotect(0x7fa5f77dd000, 8388608, PROT_READ|PROT_WRITE) = 0
clone(child_stack=0x7fa5f7fdbfb0, flags=CLONE_VM|CLONE_FS|CLONE_FILES|CLONE_SIGHAND|CLONE_THREAD|CLONE_SYSVSEM|CLONE_SETTLS|CLONE_PARENT_SETTID|CLONE_CHILD_CLEARTID, parent_tidptr=0x7fa5f7fdc9d0, tls=0x7fa5f7fdc700, child_tidptr=0x7fa5f7fdc9d0) = 60917
futex(0x7fa5fa957150, FUTEX_WAIT_PRIVATE, 0, NULL) = 0
futex(0x7fa5fa958950, FUTEX_WAKE_PRIVATE, 1) = 1
openat(AT_FDCWD, "profiles.dat", O_RDONLY) = 7
lseek(7, 0, SEEK_END)                   = 0
lseek(7, 0, SEEK_CUR)                   = 0
lseek(7, 0, SEEK_SET)                   = 0
close(7)                                = 0
--- SIGSEGV {si_signo=SIGSEGV, si_code=SEGV_MAPERR, si_addr=NULL} ---
+++ killed by SIGSEGV +++
Segmentation fault
```
looks like we got problems in the profile.dat launcher, I'll make an issue

```
d32:3ö!º<8d><8e>9êÁr<4ë=ôÑ<9e>ù²Îa<97>ù<85>ÊyÊT<8a>è·Xd1:gi1e1:pi0e1:si0e1:ti0e1:vi0ee32:=ËZ4Ð^U§»´ckè9<91>à^L¾ÿ^Sþx4àÕE/þ<9a>Zõ¥¾d1:gi1e1:pi2e1:si0e1:ti0e1:vi0ee32:YÊí^YO_cx$<99>+Æé4<93>ð¢¼Fè<8a>^T»á<9f><95>^ErÚÍ:^[d1:gi1e1:pi0e1:si0e1:ti0e1:vi0ee32:[<8e>p¬
Dhßd^F8O{]<91>^N^XWÝ´ N4Õ^yUÜ
+g©<88>d1:gi1e1:pi0e1:si0e1:ti0e1:vi0ee32:x^C<92><9d>«NMð+CðcBx^Nq±<97>Ñ^P¤ì®ÒK¦­<8e>QÝ^K<91>d1:gi1e1:pi2e1:si0e1:ti0e1:vi0eee
```

I'll just nuke my profiles.dat for now

**Comment by neuroscr** [comment]
hrm even if I delete it, I still crash

```

[router]
nickname=ldl
netid=service

[dns]
upstream=206.81.96.55
bind=127.0.0.1:53

[netdb]
dir=/root/.lokinet/netdb-service

[bootstrap]
add-node=/tmp/TqaH35w9.lokinet_signed

[network]

[api]
enabled=true
bind=127.0.0.1:28083
```

**Comment by neuroscr** [comment]
gdb:

```
0x00005555557e2ffa in bencode_read_dict (buff=0x7fffffffdc90, r=0x7fffffffdbf0) at /root/loki-daemon-launcher/src/loki-network/llarp/util/bencode.cpp:127
127	  if(*r->buffer->cur != 'd')  // ensure is a dictionary
(gdb) bt
#0  0x00005555557e2ffa in bencode_read_dict (buff=0x7fffffffdc90, r=0x7fffffffdbf0) at /root/loki-daemon-launcher/src/loki-network/llarp/util/bencode.cpp:127
#1  0x000055555558f333 in llarp::IBEncodeMessage::BDecode (this=0x555555cb56b8, buf=0x7fffffffdc90) at /root/loki-daemon-launcher/src/loki-network/llarp/util/bencode.hpp:249
#2  0x000055555578cfd0 in llarp::BDecodeReadFile<llarp::Profiling> (fpath=0x555555cb5718 "profiles.dat", t=...) at /root/loki-daemon-launcher/src/loki-network/llarp/util/bencode.hpp:307
#3  0x000055555578c6ac in llarp::Profiling::Load (this=0x555555cb56b8, fname=0x555555cb5718 "profiles.dat") at /root/loki-daemon-launcher/src/loki-network/llarp/profiling.cpp:203
#4  0x00005555555ca13e in llarp::Router::Run (this=0x555555cb2fe0, nodedb=0x555555cbecb0) at /root/loki-daemon-launcher/src/loki-network/llarp/router/router.cpp:1369
#5  0x000055555558d99c in llarp::Context::Run (this=0x555555cae470) at /root/loki-daemon-launcher/src/loki-network/llarp/context.cpp:193
#6  0x000055555558e3fe in llarp_main_run (ptr=0x555555cae450) at /root/loki-daemon-launcher/src/loki-network/llarp/context.cpp:386
#7  0x0000555555574c36 in main (argc=2, argv=0x7fffffffe588) at /root/loki-daemon-launcher/src/loki-network/daemon/main.cpp:268
```

**Comment by neuroscr** [comment]
0 byte profiles.dat caused the problem, was looking at ~/.lokinet instead of .

**Comment by majestrate** [comment]
fixed by #460
